controllers/runner_controller.go

```diff
@@ -256,6 +261,91 @@ func (r *RunnerReconciler) processRunnerCreation(ctx context.Context, runner v1a
 		return ctrl.Result{}, err
 	}
 
+	needsServiceAccount := runner.Spec.ServiceAccountName != "" && (r.UseRunnerStatusUpdateHookEphemeralRole || runner.Spec.ContainerMode == "kubernetes")
```

Comment:
I think `runner.Spec.ServiceAccountName != ""` is an indication that the user wants ARC to use the specified existing services ccount, whereas this condition results in the opposite. Perhaps we'd better invert it so that it looks like:

```suggestion
	needsServiceAccount := runner.Spec.ServiceAccountName == "" && (r.UseRunnerStatusUpdateHookEphemeralRole || runner.Spec.ContainerMode == "kubernetes")
```

Reply:
good catch - fixed in [`ec02bb7` (#1268)](https://github.com/actions-runner-controller/actions-runner-controller/pull/1268/commits/ec02bb77a613d173d79364958a19b58c1e706b0b)